lib/cn/vblock_reader.c

```diff
@@ -43,9 +44,10 @@ vbr_desc_read(
 
     vblk_desc->vbd_mblkdesc = mblk;
     vblk_desc->vbd_off = 0;
-    vblk_desc->vbd_len = wlen - VBLOCK_FOOTER_LEN;
+    vblk_desc->vbd_alen = alen - VBLOCK_FOOTER_LEN;
```

Comment:
I would expect allocated len to be the allocated size of the vblock, which would of course include the footer.